src/basic/path-util.c

```diff
@@ -719,6 +719,9 @@ const char *last_path_component(const char *path) {
         unsigned l, k;
 
         l = k = strlen(path);
+        if (l == 0) /* special case — an empty string */
+                return path;
```

Comment:
maybe simply return NULL? I mean, there is no last component in that case...

Reply:
fd_is_mount_point has `assert(filename)`, so we'd have to special case it. I think it's easier to leave it as it is now.